src/bf_functions.py

```diff
@@ -149,191 +137,150 @@ def get_bfs(self):
             inputs.append(beam_ants.index(ant_str))
-	return inputs
+        return inputs
 
     def ants2ants(self, beam, ant_strs=all):
         """expands all, None etc into valid antenna strings. Checks for valid antenna strings"""
-
         ants = []
         if ant_strs == None:
             return []
         all_ants = self.config._get_ant_mapping_list()
-
-	beam = self.beams2beams(beams=beam)[0]
-
+        beam = self.beams2beams(beams=beam)[0]
         #construct a list of valid ant_strs for each beam, then check specified ants are in
-        if self.config.simulate: print 'finding ants for beam %s' %beam
+        if self.config.simulate:
+            print 'finding ants for beam %s' %beam
         beam_ants = []
         beam_idx = self.beam2index(beam)
         offset = numpy.mod(beam_idx,2)
         for n,ant_str in enumerate(all_ants):
-	    if numpy.mod(n+offset,2) == 0:
-	        if self.config.simulate: print 'adding ant %s for beam %s' %(ant_str,beam)
-	        beam_ants.append(ant_str)
-
+            if numpy.mod(n+offset,2) == 0:
+                if self.config.simulate:
+                    print 'adding ant %s for beam %s' %(ant_str,beam)
+            beam_ants.append(ant_str)
         if ant_strs == all:
-	    return beam_ants
-
+            return beam_ants
         for ant_str in ant_strs:
-	    if(beam_ants.count(ant_str) == 0):
-	        raise fbfException(1, '%s not found in antenna mapping for beam %s'%(ant_str, beam) , \
-		                   'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-			           self.syslogger)
-	    else:
-	        ants.append(ant_str)
-
+            if(beam_ants.count(ant_str) == 0):
+                raise fbfException(1, '%s not found in antenna mapping for beam %s'%(ant_str, beam),
+                                   'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno),
+                                   self.syslogger)
+            else:
+                ants.append(ant_str)
         return ants
 
     def beams2beams(self,beams=all):
         """expands all, None etc into valid beam names. Checks for valid beam names"""
         new_beams = []
-
         if beams == None:
             return
-
         all_beams = self.get_beams()
-
         if beams == all:
             new_beams = all_beams
         else:
             if type(beams) == str:
-		beams = [beams]
-
-            #weed out beam names that do not occur
-            for beam in beams:
-                try:
-		    all_beams.index(beam)
-                    new_beams.append(beam)
-                except:
-                    raise fbfException(1, '%s not found in our system'%beam, \
-                                       'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-                                       self.syslogger)
-
+                beams = [beams]
+            # weed out beam names that do not occur
```

Comment:
You've changed the meaning here, code will now run regardless of the if statement above. Is that intended?